.github/docker-images/base-images/device-client/amazonlinux/Dockerfile

```diff
@@ -1,6 +1,6 @@
 FROM amazonlinux:2.0.20230307.0 as base
 
-ARG OPENSSL_VERSION=1.1.1n
+ARG OPENSSL_VERSION=3.1.2
```

Comment:
OpenSSL 3.0 is the LTS version which will be supported till 2026-09-07. Lets use an LTS version since it is recommended. https://www.openssl.org/policies/releasestrat.html